src/views/ProductDetails.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useEffect, useState, Fragment } from 'react';
+import { Container, Typography, Grid, Button, makeStyles, LinearProgress } from "@material-ui/core";
+import { RouteComponentProps, useHistory } from "react-router-dom";
+import { findById, deleteById } from '../data/ProductService';
+import AddShoppingCartIcon from '@material-ui/icons/AddShoppingCart';
+import DeleteForeverOutlinedIcon from '@material-ui/icons/DeleteForeverOutlined';
+import { addProductInUserCart } from '../data/CartService';
+import Product from '../domain/Product';
+import { StyledPaper } from '../components/StyledPaper';
+
+
+const useStyles = makeStyles((theme) => ({
+    img: {
+        width: '100%',
+        height: '100%'
+    },
+}));
+
+function ProductDetails(props: RouteComponentProps<any>) {
+    const params = props.match.params
+    const initialProduct = {} as Product
+
+    const [product, setProduct] = useState(initialProduct)
+    const [isLoading, setIsLoading] = useState(true)
+    const [isError, setIsError] = useState(false)
+
+    const history = useHistory();
+    const classes = useStyles();
+
+
+    useEffect(() => {
+        findById(params.id)
+            .then(result => {
+                setProduct(result.data)
+                setIsLoading(false)
+            })
+            .catch(err => {
+                setIsError(true)
+                setIsLoading(false)
+            })
+    }, [params.id]);
+
+    function addProductToCart() {
```

Comment:
Incearca sa folosesti arrow functions in loc de normal functions.
https://stackoverflow.com/questions/51778443/react-js-and-arrow-functions-vs-normal-functions
In situatia de aici cel mai probabil nu o sa faca probleme, insa sa stii pe viitor :)